### PR TITLE
Add a QuestionAnswering bert model for testing

### PR DESCRIPTION
**Pull request by kevinwuTT**
### Ticket
#10 

### Problem description
We want a separate unit test for bert. The `run_transformers.py` script has the initial implementation, and this PR separates that into a unit test. 

### What's changed
* Add a separate `test_bert.py` that runs the reference and compiled model using ttnn backend and compare the inference results.
* Remove unneeded workarounds involving type conversion. We can directly make the model use bfloat16 when setting up the model.
* Remove workaround that generated `aten.full` when scalars are passed to certain ops. Use `ttnn.full instead`
* Convert `aten._to_copy` to `ttnn.as_tensor`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -6,4 +6,5 @@ tabulate==0.9.0
 networkx==3.1
 graphviz
 matplotlib==3.7.1
+transformers==4.38.0
```
Comment (ayerofieiev-tt) [inline requirements.txt:9]: this is only needed for dev, right? As users of the lib don't need it, lets move?

File: tests/test_bert.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        # Run inference with the compiled model
+        with torch.no_grad():
+            outputs_after = m(**inputs)
+        option._out_fx_graphs[0].print_tabular()
+
+        answer_after = decode_output(outputs_after)
+
+        print(
+            f"model_name: {model_name}\ninput:\n\tcontext: {context}\n\tquestion: {question}\nanswer before: {answer_before}\nanswer_after: {answer_after}"
+        )
```
Comment (ayerofieiev-tt) [inline tests/test_bert.py:74]: Consider
```suggestion
print(f"""
model_name: {model_name}
input:
    context: {context}
    question: {question}
answer before: {answer_before}
answer after: {answer_after}
""")

```

File: tests/test_more_ops.py [hunk 3]
```diff
@@ -555,6 +555,29 @@ def input_shapes(self):
         return [(4, 4)]
 
 
+class ToCopyModule(torch.nn.Module):
```
Comment (ayerofieiev-tt) [inline tests/test_more_ops.py:558]: We really have to start splitting things into more logical files. 
"more_ops" is.. not good

File: torch_ttnn/passes/to_tt_pass.py [hunk 4]
```diff
@@ -300,42 +300,74 @@ def ReplaceMoreTtManually(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
...
                 # NOTE(kevinwuTT): Only bfloat16 seems to work for now
                 # TODO(kevinwuTT): Use ttnn.full instead of aten
-                new_kwargs = {"dtype": torch.bfloat16}
-                full_node = g.call_function(
-                    torch.ops.aten.full.default,
-                    args=(arg_metadata.size(), args[1]),
-                    kwargs=new_kwargs,
+                new_kwargs = {
+                    "fill_value": args[1],
+                    "device": DummyDevice(),
```
Comment (ayerofieiev-tt) [inline torch_ttnn/passes/to_tt_pass.py:309]: Why do we need this?

File: torch_ttnn/passes/to_tt_pass.py [hunk 5]
```diff
@@ -300,42 +300,74 @@ def ReplaceMoreTtManually(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
...
+                kwargs = node.kwargs
+                dummy_dtype = torch_dtype_to_dummy_ttnn_dtype(kwargs["dtype"])
+                # Add additional logic to choose the appropriate memory_config type: DRAM or L1
+                memory_config = g.call_function(
+                    ttnn.MemoryConfig,
+                    (
+                        DummyTtlTensorTensorMemoryLayoutInterleaved(),
+                        DummyTtlTensorBufferTypeDram(),
+                    ),
+                )
```
Comment (ayerofieiev-tt) [inline torch_ttnn/passes/to_tt_pass.py:359]: Can you help to understand whats going on here?

**Review by ayerofieiev-tt — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (kevinwuTT) [reply requirements.txt]: Done.

File: tests/test_more_ops.py (hunk 3 above)
Comment (kevinwuTT) [reply tests/test_more_ops.py:558]: Yeah will definitely prioritize this during refactoring soon.

**Pull request merged**